### PR TITLE
feat: create new directory

### PR DESCRIPTION
**Pull request by Woynert**
## Includes

- Create new directory SOAP controller implementation
- Updated Open API specification
- Update python cli
- Add extra tests for uploading files in directories

## How to try

Make sure all services all running:

```sh
docker-compose up
```

Then run all tests:

```sh
./gradlew cleanTest test
```

### CODE REVIEW

**Comment by SilviaPabon** [comment]
oh wow, someone steal my stuff 🤯

File: app/src/test/java/gateway/ITFileIO.java [hunk 1]
```diff
@@ -90,7 +92,43 @@ static class State
 		assertEquals (500, CtrlFileUpload.file_upload (args).code, "Can't reach Auth");
 	}
 
-	@Test @Order (2) void file_check () throws InterruptedException
+	@Test @Order (3) void uploadFileOnDirectory ()
```
Comment (SilviaPabon) [inline app/src/test/java/gateway/ITFileIO.java:95]: In this test there is not code of error? Like 400, 409...? 🤔

File: app/src/test/java/gateway/ITFileManagment.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+		TestUtilConfig.makeInvalidMetadata ();
+		assertEquals (500, CtrlFileNewDir.file_new_dir (reqD).code, "Can't reach metadata");
+
+		reqD.token = "invalid-token";
+		assertEquals (401, CtrlFileNewDir.file_new_dir (reqD).code, "Unauthorized");
+
+		reqD.directoryName = null;
+		assertEquals (400, CtrlFileNewDir.file_new_dir (reqD).code, "Field validation failed");
+	}
```
Comment (SilviaPabon) [inline app/src/test/java/gateway/ITFileManagment.java:57]: What about 409 response? 🤔

File: app/src/main/java/gateway/services/ServiceMetadata.java [hunk 3]
```diff
@@ -17,16 +17,17 @@ public static class ResSaveFile extends ResStatus
...
 
 		ServiceMetadata.ResSaveFile s = new ServiceMetadata.ResSaveFile ();
 		JSONObject body = new JSONObject ();
 
 		body.put ("userUUID", userUUID);
 		body.put ("parentUUID", directoryUUID == null ? JSONObject.NULL : directoryUUID);
-		body.put ("fileType", "archive");
+		body.put ("fileType", isFile ? "archive" : "directory");
 		body.put ("fileName", filename);
 		body.put ("fileExtension", filetype == null ? JSONObject.NULL : filetype);
```
Comment (SilviaPabon) [inline app/src/main/java/gateway/services/ServiceMetadata.java:32]: I suggest you to change fileType for fileExtension in this case, what do you think?

**Review by SilviaPabon — CHANGES_REQUESTED** [review]

File: app/src/test/java/gateway/ITFileIO.java (hunk 1 above)
Comment (Woynert) [reply app/src/test/java/gateway/ITFileIO.java:95]: This test exclusively tests the case when a file is uploaded in a directory.

Tests for the create directory endpoint are located in `ITFileManagment.java`.

**Comment by PedroChaparro** [comment]
Could you create a release when this Pull Request gets merged?

**Review by SilviaPabon — APPROVED** [review]

**Pull request merged**